### PR TITLE
Adding a new metadata payload for system-probe

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

For now the payload only contains the agent version and the different layers of configuration for system-probe.

### Describe how to test/QA your changes

Check that:
- Configuration from system-probe is accurate in the payload
- the `diagnose show-metadata sytem-probe` cli command works
- The system-probe.yaml file is correctly added to the flare

### CODE REVIEW

File: comp/metadata/systemprobe/impl/system_probe_test.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	authtokenimpl "github.com/DataDog/datadog-agent/comp/api/authtoken/fetchonlyimpl"
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/core/log/logimpl"
+	"github.com/DataDog/datadog-agent/comp/core/sysprobeconfig"
+	"github.com/DataDog/datadog-agent/comp/core/sysprobeconfig/sysprobeconfigimpl"
+	configFetcher "github.com/DataDog/datadog-agent/pkg/config/fetcher"
+	"github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/DataDog/datadog-agent/pkg/serializer"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
```
Comment (GustavoCaso) [inline comp/metadata/systemprobe/impl/system_probe_test.go:30]: I guess we need to decide what to do with testing for the implementation 😄

File: comp/metadata/systemprobe/impl/system_probe.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+	"github.com/DataDog/datadog-agent/comp/api/api"
+	"github.com/DataDog/datadog-agent/comp/api/api/utils"
+	"github.com/DataDog/datadog-agent/comp/api/authtoken"
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	flaretypes "github.com/DataDog/datadog-agent/comp/core/flare/types"
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/core/sysprobeconfig"
+	"github.com/DataDog/datadog-agent/comp/metadata/internal/util"
+	"github.com/DataDog/datadog-agent/comp/metadata/runner/runnerimpl"
+	"github.com/DataDog/datadog-agent/comp/metadata/systemprobe/def"
```
[the remark this answers is not in the document]
Comment (GustavoCaso) [reply comp/metadata/systemprobe/impl/system_probe.go:25]: Or `systemprobemetadata "github.com/DataDog/datadog-agent/comp/metadata/systemprobe/def"`

**Review by jeremy-hanna — APPROVED** [review]
👍 LGTM, nice work!

File: comp/metadata/systemprobe/impl/system_probe_test.go (hunk 1 above)
Comment (hush-hush) [reply comp/metadata/systemprobe/impl/system_probe_test.go:30]: Yes, having ways to get mock without FX is going to be critical.

File: comp/metadata/systemprobe/impl/system_probe.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (hush-hush) [reply comp/metadata/systemprobe/impl/system_probe.go:25]: Why not `def` ?

[the remark this answers is not in the document]
Comment (GustavoCaso) [reply comp/metadata/systemprobe/impl/system_probe.go:25]: It seems that is not consistent across your PR how we import this package. Also, I find `def.Component` less clear than `systemprobemetadata.Component` or `systemprobe.Component`

**Pull request merged**